### PR TITLE
feat: Add app mode operational status logic

### PR DESCRIPTION
**Pull request by lsetiawan**
This PR depends on #95. It adds an additional state attribute for `AppState` called `mode`. This is meant to keep track of the `operational_status` of the app. There are 3 operational status of `emergency`, `normal`, and `test` mode. This PR relates to #27.

### CODE REVIEW

File: deployment/values.dev.yaml [hunk 1]
```diff
@@ -83,6 +83,10 @@ rest:
     tag: 2024.09.13
 
 realtime:
+  # Override the fullname of the realtime deployment
+  # as this is needed by supabase realtime to work properly
+  # See: https://github.com/supabase-community/supabase-kubernetes/issues/74
+  fullnameOverride: "realtime-dev"
```
Comment (nikiburggraf) [inline deployment/values.dev.yaml:89]: Should we change this in values.cloud.yaml too?

**Review by nikiburggraf — APPROVED** [review]
One question for ya, happy for it to be addressed in a follow-up

File: deployment/values.dev.yaml (hunk 1 above)
Comment (lsetiawan) [reply deployment/values.dev.yaml:89]: Yes! Would you be able to do that in separate PR since you have quick decrypt access? Thank you!

Comment (nikiburggraf) [reply deployment/values.dev.yaml:89]: Can do!

Comment (lsetiawan) [reply deployment/values.dev.yaml:89]: Thanks so much!

Comment (nikiburggraf) [reply deployment/values.dev.yaml:89]: https://github.com/uw-ssec/post-disaster-comms/pull/105

**Pull request merged**